### PR TITLE
Consider alpha to coverage in FS output interface validation

### LINKED ISSUE #703: FS output interface validation should consider alphaToCoverage state
Alpha to coverage uses location 0 alpha component.

- When using alpha to coverage, we should not warn on a "useless" declaration of this output.
- When using alpha to coverage, if location 0 alpha component is *not* declared, that should be an error.

### PR DESCRIPTION
**Pull request by asuonpaa**
Alpha to coverage uses location 0 alpha component.

* When using alpha to coverage, we should not warn on a "useless" declaration of this output.
* When using alpha to coverage, if location 0 alpha component is _not_ declared, that should be an error.

Fixes #703

### CODE REVIEW

**Comment by mark-lunarg** [comment]
Hi @asuonpaa, LGTM.  It also looks like these commits should be squashed?  And can you take a look at the 'Show all checks' link above, then at the 'Check Commit Format' results on the Travis CI build? This formatting helps to keep things consistent in the repo.  Thanks!

**Comment by jzulauf-lunarg** [comment]
@asuonpaa -- marking WIP, need to resolve issue on internal CI

**Comment by Tony-LunarG** [comment]
This is getting a "map/set iterator not dereferencable" at the if statement here:
```
while ((outputs.size() > 0 && it_a != outputs.end()) || (color_attachments.size() > 0 && it_b != color_attachments.end())) {
        bool a_at_end = outputs.size() == 0 || it_a == outputs.end();
        bool b_at_end = color_attachments.size() == 0 || it_b == color_attachments.end();

        if (it_a->first.first == 0 && fs->get_def(it_a->second.type_id) != fs->end() &&
            GetComponentsConsumedByType(fs, it_a->second.type_id, false) == 4)
```
in DeviceCanDraw with this shader:
```
   // Draw with a solid color.
    char const *fsSource =
        "#version 450\n"
        "\n"
        "layout(location=0) out vec4 color;\n"
        "void main() {\n"
        "   color = vec4(32.0/255.0);\n"
        "}\n";
```

File: layers/shader_validation.cpp [hunk 1]
```diff
@@ -949,17 +949,26 @@ static bool ValidateFsOutputsAgainstRenderPass(debug_report_data const *report_d
...
+                                  pipeline->graphicsPipelineCI.pMultisampleState->alphaToCoverageEnable == VK_TRUE;
+    bool locationZeroHasAlpha = false;
 
     // Walk attachment list and outputs together
 
     while ((outputs.size() > 0 && it_a != outputs.end()) || (color_attachments.size() > 0 && it_b != color_attachments.end())) {
         bool a_at_end = outputs.size() == 0 || it_a == outputs.end();
         bool b_at_end = color_attachments.size() == 0 || it_b == color_attachments.end();
 
+        if (it_a->first.first == 0 && fs->get_def(it_a->second.type_id) != fs->end() &&
```
Comment (jzulauf-lunarg) [inline layers/shader_validation.cpp]: this is potentially dereferencing outputs.end(), checked above and below, but not here.

**Comment by jzulauf-lunarg** [comment]
Removing WIP as revisions eliminate crashes in internal CI.

**Comment by jzulauf-lunarg** [comment]
@asuonpaa  -- did you note the request to break this into two commits?  Don't want to hold this up further.

**Comment by asuonpaa** [comment]
Is there any action required from me?

**Comment by jzulauf-lunarg** [comment]
@asuonpaa -- sorry, went down a design rabbit hole :).

**Pull request merged**